source/socket.c

```diff
@@ -0,0 +1,259 @@
+}
+
+int aws_socket_stop_accept(struct aws_socket *socket) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_stop_accept_fn);
+    return socket->vtable->socket_stop_accept_fn(socket);
+}
+
+int aws_socket_close(struct aws_socket *socket) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_close_fn);
+    return socket->vtable->socket_close_fn(socket);
+}
+
+int aws_socket_shutdown_dir(struct aws_socket *socket, enum aws_channel_direction dir) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_shutdown_dir_fn);
+    return socket->vtable->socket_shutdown_dir_fn(socket, dir);
+}
+
+int aws_socket_set_options(struct aws_socket *socket, const struct aws_socket_options *options) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_set_options_fn);
+    return socket->vtable->socket_set_options_fn(socket, options);
+}
+
+int aws_socket_assign_to_event_loop(struct aws_socket *socket, struct aws_event_loop *event_loop) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_assign_to_event_loop_fn);
+    return socket->vtable->socket_assign_to_event_loop_fn(socket, event_loop);
+}
+
+struct aws_event_loop *aws_socket_get_event_loop(struct aws_socket *socket) {
+    return socket->event_loop;
+}
+
+int aws_socket_subscribe_to_readable_events(
+    struct aws_socket *socket,
+    aws_socket_on_readable_fn *on_readable,
+    void *user_data) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_subscribe_to_readable_events_fn);
+    return socket->vtable->socket_subscribe_to_readable_events_fn(socket, on_readable, user_data);
+}
+
+int aws_socket_read(struct aws_socket *socket, struct aws_byte_buf *buffer, size_t *amount_read) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_read_fn);
+    return socket->vtable->socket_read_fn(socket, buffer, amount_read);
+}
+
+int aws_socket_write(
+    struct aws_socket *socket,
+    const struct aws_byte_cursor *cursor,
+    aws_socket_on_write_completed_fn *written_fn,
+    void *user_data) {
+
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_write_fn);
+    return socket->vtable->socket_write_fn(socket, cursor, written_fn, user_data);
+}
+
+int aws_socket_get_error(struct aws_socket *socket) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_get_error_fn);
+    return socket->vtable->socket_get_error_fn(socket);
+}
+
+bool aws_socket_is_open(struct aws_socket *socket) {
+    AWS_PRECONDITION(socket->vtable && socket->vtable->socket_is_open_fn);
+    return socket->vtable->socket_is_open_fn(socket);
+}
+
+/**
+ * Return the default socket implementation type. If the return value is `AWS_SOCKET_IMPL_PLATFORM_DEFAULT`, the
+ * function failed to retrieve the default type value.
+ */
+static enum aws_socket_impl_type aws_socket_get_default_impl_type(void) {
+    enum aws_socket_impl_type type = AWS_SOCKET_IMPL_PLATFORM_DEFAULT;
+// override default socket
+#ifdef AWS_USE_APPLE_NETWORK_FRAMEWORK
```

Comment:
This is pretty convoluted.  Can we just simplify to
```
#ifdef AWS_USE_APPLE_NETWORK_FRAMEWORK
static enum aws_socket_impl_type aws_socket_get_default_impl_type(void) { 
  return AWS_SOCKET_IMPL_APPLE_NETWORK_FRAMEWORK;
}
#else 
<... normal impl>
#endif
```

I don't see how we're not getting dead code warnings when AWS_USE_APPLE_NETWORK is defined.